v1/contracts/VotingToken.sol

```diff
@@ -0,0 +1,33 @@
+pragma solidity ^0.5.0;
+
+import "./MultiRole.sol";
+
+import "openzeppelin-solidity/contracts/drafts/ERC20Snapshot.sol";
+
+
+/**
+ * @title UMA voting token
+ * @dev Supports snapshotting and allows the Oracle to mint new tokens as rewards.
+ */
+contract VotingToken is ERC20Snapshot, MultiRole {
+
+    enum Roles {
+        // Can set the minter.
+        Governance,
+        // The Oracle contract (currently named Voting.sol) can mint new tokens as voting rewards.
+        Minter
+    }
+
+    constructor() public {
+        _createExclusiveRole(uint(Roles.Governance), uint(Roles.Governance), msg.sender);
+        _createExclusiveRole(uint(Roles.Minter), uint(Roles.Governance), msg.sender);
+    }
+
+    /**
+     * @dev Mints `value` tokens to `recipient`, returning true on success.
+     */
+    function mint(address recipient, uint value) external onlyRoleHolder(uint(Roles.Minter)) returns (bool) {
```

Comment:
What about an accompanying burn method?

Reply:
Done. I wasn't sure whether the `burn` method should be permissioned or open to everyone. I went with permissioned, figuring that we can always add a wrapper contract that can burn tokens that exposes an unpermissioned burn method to open up the functionality.